core/node/events/snapshot.go

```diff
@@ -191,14 +191,51 @@ func update_Snapshot_Space(iSnapshot *Snapshot, spacePayload *SpacePayload, even
 	case *SpacePayload_Inception_:
 		return RiverError(Err_INVALID_ARGUMENT, "cannot update blockheader with inception event")
 	case *SpacePayload_Channel:
+		channelStreamId, err := shared.StreamIdFromBytes(content.Channel.ChannelId)
+		if err != nil {
+			return RiverError(Err_INVALID_ARGUMENT, "invalid channel id").
+				Tag("channelId", content.Channel.ChannelId)
+		}
 		channel := &SpacePayload_ChannelMetadata{
 			ChannelId:         content.Channel.ChannelId,
 			Op:                content.Channel.Op,
 			OriginEvent:       content.Channel.OriginEvent,
 			UpdatedAtEventNum: eventNum,
 		}
+		// Apply default channel settings on creation
+		if channel.Op == ChannelOp_CO_CREATED {
+			channel.Autojoin = shared.IsDefaultChannelId(channelStreamId)
+			channel.ShowUserJoinLeaveEvents = true
+		}
+
```

Comment:
we need to update the existing channel metadata here otherwise we lose the new properties